<a href="https://colab.research.google.com/github/Ethan0625/Study_room/blob/main/Pytorch/Pandas_Exercise.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

In [2]:
# 데이터 불러오기
df = pd.read_csv('/content/drive/MyDrive/Study/Pytorch/PYTORCH_NOTEBOOKS/00-Crash-Course-Topics/01-Crash-Course-Pandas/bank.csv')

In [3]:
# 5개의 row만 출력하기
df.head()

age          job  marital  education  ... pdays  previous poutcome   y
0   30   unemployed  married    primary  ...    -1         0  unknown  no
1   33     services  married  secondary  ...   339         4  failure  no
2   35   management   single   tertiary  ...   330         1  failure  no
3   30   management  married   tertiary  ...    -1         0  unknown  no
4   59  blue-collar  married  secondary  ...    -1         0  unknown  no

[5 rows x 17 columns]

In [4]:
# 데이터의 age 칼럼 평균 구하기
df['age'].mean()

41.17009511170095

In [5]:
# 가장 나이가 어린 사람의 혼인여부 찾기
df['marital'][df['age'].idxmin()]

'single'

In [6]:
# 풀이
# 가장 나이가 어린 사람 찾기
df['age'].min()

19

In [10]:
df[df['age']==19]

age      job marital  education  ... pdays  previous poutcome    y
503    19  student  single    primary  ...    -1         0  unknown  yes
1900   19  student  single    unknown  ...    -1         0  unknown   no
2780   19  student  single  secondary  ...    -1         0  unknown  yes
3233   19  student  single    unknown  ...    -1         0  unknown   no

[4 rows x 17 columns]

가장 나이가 어린 사람 모두 single인 상태임.  
그중 제일 첫 번째 사람의 index는 503.  
따라서 'marital'칼럼에서 503번째 사람을 return하면 정답임.

In [11]:
df['marital'][503]

'single'

503이라는 index는 위의 과정을 거쳐 찾을 수도 있지만, idxmin()이라는 메소드를 통해서도 찾을 수 있음.

In [12]:
df['marital'][df['age'].idxmin()]

'single'

In [13]:
# 직업 카테고리의 유일 갯수 구하기
df['job'].nunique()

12

In [14]:
# 각 직업군에 해당하는 인원이 몇명인지 구하기
df['job'].value_counts()

management       969
blue-collar      946
technician       768
admin.           478
services         417
retired          230
self-employed    183
entrepreneur     168
unemployed       128
housemaid        112
student           84
unknown           38
Name: job, dtype: int64

In [16]:
# 데이터셋에서 결혼 한 사람의 비율 구하기
(len(df[df['marital']=='married'])/len(df))*100

61.86684361866843

In [17]:
# 풀이
# 'marital'칼럼에서 'married' 즉 결혼한 인원을 조건으로 만듦
df['marital']=='married'

0        True
1        True
2       False
3        True
4        True
        ...  
4516     True
4517     True
4518     True
4519     True
4520    False
Name: marital, Length: 4521, dtype: bool

조건식이 되었기 때문에 이를 df[]안에 넣어주면 조건에 해당되는 인원만 표기 가능

In [18]:
df[df['marital']=='married']

age            job  marital  education  ... pdays  previous poutcome   y
0      30     unemployed  married    primary  ...    -1         0  unknown  no
1      33       services  married  secondary  ...   339         4  failure  no
3      30     management  married   tertiary  ...    -1         0  unknown  no
4      59    blue-collar  married  secondary  ...    -1         0  unknown  no
6      36  self-employed  married   tertiary  ...   330         2    other  no
...   ...            ...      ...        ...  ...   ...       ...      ...  ..
4514   38    blue-collar  married  secondary  ...   153         1  failure  no
4516   33       services  married  secondary  ...    -1         0  unknown  no
4517   57  self-employed  married   tertiary  ...    -1         0  unknown  no
4518   57     technician  married  secondary  ...    -1         0  unknown  no
4519   28    blue-collar  married  secondary  ...   211         3    other  no

[2797 rows x 17 columns]

총 2797개의 row를 가지게 되었으므로, 이를 len()을 활용하여 갯수로 변환함.

In [19]:
len(df[df['marital']=='married'])

2797

데이터 셋의 총 인원 중 결혼한 인원의 갯수를 확인 하였으니, 이를 총 인원수로 나누어 비율을 계산함.

In [20]:
100*len(df[df['marital']=='married'])/len(df)

61.86684361866843

In [22]:
# .apply를 활용하여 기혼 여부를 's' 또는 'm'으로 나타내는 칼럼 'marital code' 만들기
df['marital code']=df['marital'].apply(lambda status: status[0])
df

age            job  marital  ... poutcome   y  marital code
0      30     unemployed  married  ...  unknown  no             m
1      33       services  married  ...  failure  no             m
2      35     management   single  ...  failure  no             s
3      30     management  married  ...  unknown  no             m
4      59    blue-collar  married  ...  unknown  no             m
...   ...            ...      ...  ...      ...  ..           ...
4516   33       services  married  ...  unknown  no             m
4517   57  self-employed  married  ...  unknown  no             m
4518   57     technician  married  ...  unknown  no             m
4519   28    blue-collar  married  ...    other  no             m
4520   44   entrepreneur   single  ...    other  no             s

[4521 rows x 18 columns]

In [23]:
# 'duration' 칼럼에서 가장 큰 값 찾기
df['duration'].max()

3025

In [24]:
# 직업이 없는 사람들의 교육 현황 구하기
df[df['job']=='unemployed']['education'].value_counts()

secondary    68
tertiary     32
primary      26
unknown       2
Name: education, dtype: int64

In [25]:
# 직업이 없는 사람들의 나이 평균 구하기
df[df['job']=='unemployed']['age'].mean()

40.90625